### PR TITLE
story(ccmspui-313) enable general request submission

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**

### CODE REVIEW

**Review by JamieBriggs-MoJ — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java [hunk 1]
```diff
@@ -77,7 +81,9 @@ public class ProviderRequestsController {
 
   private final ProviderRequestsMapper providerRequestsMapper;
 
-  private final AvScanService avScanService;
+  private static final String UNRELATED_CASE_REFERNCE = "-1";
```
Comment (farrell-m) [inline src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java:84]:
```suggestion
  private static final String UNRELATED_CASE_REFERENCE = "-1";
```

File: src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java [hunk 2]
```diff
@@ -245,8 +251,35 @@ public String postRequestDetail(
...
+                    documentSessionId,
+                    CcmsModule.REQUEST)
+                .blockOptional()
+                .orElseThrow(() -> new CaabApplicationException(
+                    String.format("Invalid document session id: %s", documentSessionId)));
+
+        //using the list of documents, we need to upload them to ebs through soa.
+        // we also update the status of the documents in the tds to say they have been uploaded
+        evidenceService.uploadAndUpdateDocuments(
+            documents, UNRELATED_CASE_REFERNCE, notificationId, userDetail).block();
```
Comment (farrell-m) [inline src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java:280]:
```suggestion
            documents, UNRELATED_CASE_REFERENCE, notificationId, userDetail).block();
```

File: src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java [hunk 3]
```diff
@@ -292,7 +325,7 @@ public String addDocumentsToRequest(
 
     //set the additional details for the evidence upload
     evidenceUploadFormData.setApplicationOrOutcomeId(documentSessionId);
-    evidenceUploadFormData.setCaseReferenceNumber("-1");
+    evidenceUploadFormData.setCaseReferenceNumber(UNRELATED_CASE_REFERNCE);
```
Comment (farrell-m) [inline src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java:328]:
```suggestion
    evidenceUploadFormData.setCaseReferenceNumber(UNRELATED_CASE_REFERENCE);
```

File: src/test/java/uk/gov/laa/ccms/caab/mapper/ProviderRequestsMapperTest.java
Comment (farrell-m) [inline src/test/java/uk/gov/laa/ccms/caab/mapper/ProviderRequestsMapperTest.java:1]: Was this no longer needed?

Comment (PhilDigitalJustice) [reply src/test/java/uk/gov/laa/ccms/caab/mapper/ProviderRequestsMapperTest.java:1]: ah missed this file, will write up some tests

File: src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java (hunk 1 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java:84]: updated

File: src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java (hunk 3 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java:328]: updated

**Comment by PhilDigitalJustice** [comment]
Dependant on: https://github.com/ministryofjustice/laa-ccms-caab/pull/316
Will need to amend new buttons introduced

**Pull request merged**